src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,149 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents a list of tasks.
+ * Chatbot users can perform add/delete/mark operations on the tasks in the list.
+ */
+public class TaskList {
+    private List<Task> tasks;
+
+    public TaskList(List<Task> l) {
+        this.tasks = l;
+    }
+
+    public TaskList(TaskList l) {
+        this.tasks = l.tasks;
+    }
+
+    public TaskList() {
+        this.tasks = new ArrayList<Task>();
+    }
```

Comment:
You can consider removing the use of `this` here (and elsewhere in the class) since the `tasks` variable is not shadowing any other variables. However, this is an 'advanced' coding standard rule so you do not need to follow it.

Reply:
Yes I agree with this! Might be good to standardize the style used throughout too, such as in the getTask() method, where "this" was omitted.

Reply:
Thanks! I will standardize the style by remove `this`